storage/storage.rs

```diff
@@ -452,31 +279,23 @@ impl<D> MVCCStorage<D> {
         }
 
         let previous_checkpoints: Vec<_> = fs::read_dir(&checkpoint_dir)
-            .map_err(|error| ReadCheckpointDir { dir: checkpoint_dir.clone(), source: error })?
-            .map_ok(|entry| entry.path())
-            .try_collect()
+            .and_then(|entries| entries.map_ok(|entry| entry.path()).try_collect())
             .map_err(|error| ReadCheckpointDir { dir: checkpoint_dir.clone(), source: error })?;
 
         let current_checkpoint_dir = checkpoint_dir.join(format!("{}", Utc::now().timestamp_micros()));
```

Comment:
I think creating a checkpoint directory name and reading from a checkpoint directory name string to a UTC timestamp is critical enough we should create methods for these steps